src/NotesLoaderSM.cpp

```diff
@@ -979,26 +979,57 @@ bool SMLoader::LoadFromBGChangesString( BackgroundChange &change, const RString
 		// Backward compatibility:
 		if( change.m_def.m_sEffect.empty() )
 		{
-			bool bLoop = std::stoi( aBGChangeValues[5] ) != 0;
-			if( !bLoop )
+			try
+			{
+				if( std::stoi( aBGChangeValues[5] ) != 0 )
+				{
+					change.m_def.m_sEffect = SBE_StretchNoLoop;
+				}
+			}
+			catch ( const std::invalid_argument & )
+			{
+				// Failed conversions mean it's not 0 as well
```

Comment:
Does the exception really mean it isn't 0? The old C atoi function would return 0 if it wasn't able to decode a number at all from the provided string.

Reply:
The exception is thrown if no conversion could be performed. That means the string could not have been interpreted as 0. According to the spec this is the correct behavior.